samza-core/src/main/java/org/apache/samza/clustermanager/ClusterBasedJobCoordinator.java

```diff
@@ -368,8 +368,8 @@ void generateAndUpdateJobCoordinatorMetadata(JobModel jobModel) {
     JobCoordinatorMetadataManager jobCoordinatorMetadataManager = createJobCoordinatorMetadataManager();
 
     JobCoordinatorMetadata previousMetadata = jobCoordinatorMetadataManager.readJobCoordinatorMetadata();
-    JobCoordinatorMetadata newMetadata = jobCoordinatorMetadataManager.generateJobCoordinatorMetadata(jobModel, config);
```

Comment:
There are scenarios where the job model object doesn't have config populated within. E.g., Standalone.

Hence the signature to explicitly pass around config. IMO, `JobModel` being a container for configuration shouldn't be the ideal case and I had to keep this around for back-compat reasons in the YARN (a single servlet to expose the configuration + JobModel to workers).

That said, the current model  constrains updates to one vs another and forces to update the container object `JobModel` as a whole. In YARN, this isn't a problem, because the lifecycle of both co-exist. 

Do you have any strong reasons behind making this change? If not, I'd  prefer to keep it the way it is so that it allows us the flexibility to not depend on the config within `JobModel` and potentially help in the path of extracting it from `JobModel`.

Reply:
Oh, that's pretty misleading that config isn't always fully filled in, and it's worse because now `JobModel` is part of the API.
I thought I was just doing a minor clean-up, so this doesn't need to be changed.
Regarding your note about `JobModel` being a container for config: `JobModel` is the single object which gets coordinated across containers, and it still feels like config might be something that should be coordinated between containers, so I'm not totally convinced that config should be pulled out of `JobModel`. Anyways, that is out-of-scope for this PR.